core/components/organisms/table/Header.tsx

```diff
@@ -209,11 +210,19 @@ export const Header = (props: HeaderProps) => {
               {withPagination && selectAll?.checked && allowSelectAll && (
                 <div className="ml-4">
                   {!selectAllRecords ? (
-                    <Button size="tiny" onClick={() => setSelectAllRecords(true)}>
+                    <Button
+                      size="tiny"
+                      data-test="DesignSystem-Header-selectAllItemsButton"
```

Comment:
please change the test id to:- `DesignSystem-Table-Header--selectAllItemsButton`